File: 2021-02/javascript/rotational-cipher/rotational-cipher.js
export class RotationalCipher {
  static rotate(text, key) { 

    let secretMessage = ''
    for (const char of text) {
      secretMessage += this.cipher(char, key)
    }
    return secretMessage
  }

  static cipher(char, key) {
    if (('a'.charCodeAt(0) <= char.charCodeAt(0)) && (char.charCodeAt(0) <= 'z'.charCodeAt(0))) {
      console.log(char)
      const code = (char.charCodeAt(0) - 'a'.charCodeAt(0) + key) % 26 + 'a'.charCodeAt(0)
      return String.fromCharCode(code);
    }
    if (('A'.charCodeAt(0) <= char.charCodeAt(0)) && (char.charCodeAt(0) <= 'Z'.charCodeAt(0))) {
      const code = (char.charCodeAt(0) - 'A'.charCodeAt(0) + key) % 26 + 'A'.charCodeAt(0)
      return String.fromCharCode(code)
    }

    return char
  }

}

File: 2021-02/python/rotational-cipher/rotational_cipher.py
def rotate(text, key):

  return "".join([cipher(char,key) for char in text])

  # Alternative using a temp variable and a for loop

  #secret_message = ""

  #for char in text:
  #  secret_message += cipher(char,key)

  #return secret_message 

def cipher(char,key):
    # if it's in ascii range 65 - 90 (uppercase) or 97 - 122 (lowercase) then its a letter
    if ord('A')<=ord(char)<=ord('Z'):
      return chr((ord(char)-ord('A')+key)%26+ord('A'))
    if ord('a')<=ord(char)<=ord('z'):
      return chr((ord(char)-ord('a')+key)%26+ord('a'))
    return char


